sympy/sets/sets.py

```diff
@@ -1438,6 +1439,44 @@ def _handle_finite_sets(args):
             return
         s = fs_args[0]
         fs_args = fs_args[1:]
+        if len(args) == 2 and len(s) == 1 and len(new_args) == 1 and isinstance(new_args[0], Interval):
+            try:
+                ineq = s.__iter__().__next__()
+            except AttributeError:
+                ineq = s.__iter__().next()
```

Comment:
regarding q, yes it should not be needed, I will modify it.
regarding next function, I will check again.